src/main/java/com/plaid/client/request/LinkTokenCreateRequest.java

```diff
@@ -0,0 +1,172 @@
+import com.plaid.client.request.common.BaseClientRequest;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class LinkTokenCreateRequest extends BaseClientRequest {
+  private User user;
+  private String clientName;
+  private List<String> products;
+  private List<String> countryCodes;
+  private String language;
+  private String webhook;
+  private String accessToken;
+  private String linkCustomizationName;
+  private String redirectUri;
+  private String androidPackageName;
+  private Map<String, SubtypeFilters> accountFilters;
+  private LinkTokenCreateRequest.CrossAppItemAdd crossAppItemAdd;
+  private String paymentInitiation;
+
+  public LinkTokenCreateRequest(
+    User user,
+    String clientName,
+    List<String> products
+  ) {
+    this.user = user;
+    this.clientName = clientName;
+    this.products = products;
+  }
+
+  public static class SubtypeFilters {
+    private List<String> accountSubtypes;
+    private List<String> subtypes;
+
+    public SubtypeFilters(List<String> accountSubtypes) {
+      this.accountSubtypes = accountSubtypes;
+    }
+  }
+
+  public static class User {
+    private String clientUserId;
+    private String legalName;
+    private String phoneNumber;
+    private String emailAddress;
+    @JsonAdapter(OptionalDateTypeAdapterFactory.class)
+    private Optional<Date> phoneNumberVerifiedTime = Optional.empty();
+    @JsonAdapter(OptionalDateTypeAdapterFactory.class)
+    private Optional<Date> emailAddressVerifiedTime = Optional.empty();
+
+    public User(String clientUserId) {
+      this.clientUserId = clientUserId;
+    }
+
+    public User withClientUserId(String clientUserId) {
+      this.clientUserId = clientUserId;
+      return this;
+    }
+
+    public User withLegalName(String legalName) {
+      this.legalName = legalName;
+      return this;
+    }
+
+    public User withPhoneNumber(String phoneNumber) {
+      this.phoneNumber = phoneNumber;
+      return this;
+    }
+
+    public User withEmailAddress(String emailAddress) {
+      this.emailAddress = emailAddress;
+      return this;
+    }
+
+    public User withPhoneNumberVerifiedTime(Date phoneNumberVerifiedTime) {
+      if (phoneNumberVerifiedTime == null) {
+        this.phoneNumberVerifiedTime = Optional.empty();
+      } else {
+        this.phoneNumberVerifiedTime = Optional.of(phoneNumberVerifiedTime);
+      }
+      return this;
+    }
+
+    public User withEmailAddressVerifiedTime(Date emailAddressVerifiedTime) {
+      if (emailAddressVerifiedTime == null) {
+        this.emailAddressVerifiedTime = Optional.empty();
+      } else {
+        this.emailAddressVerifiedTime = Optional.of(emailAddressVerifiedTime);
+      }
+      return this;
+    }
+  }
+
+  public static class CrossAppItemAdd {
+    private String targetApplicationToken;
+    private String foreignId;
+
+    public CrossAppItemAdd(String targetApplicationToken, String foreignId) {
```

Comment:
I don't think foreignId is required. let's make this consistent with the other classes and add a `withForeignId` here.